contract/src/airdrop/prepare.js

```diff
@@ -0,0 +1,267 @@
+import { M } from '@endo/patterns';
+import { makeDurableZone } from '@agoric/zone/durable.js';
+import { E } from '@endo/far';
+import { Far } from '@endo/marshal';
+import { AmountMath } from '@agoric/ertp';
+import { TimeMath, TimestampShape } from '@agoric/time';
+import '../../types.js'
+
+/**
+ * @typedef {object} EpochDetails
+ * @property {bigint} windowLength Length of epoch in seconds. This value is used by the contract's timerService to schedule a wake up that will fire once all of the seconds in an epoch have elapsed
+ * @property {bigint} earlyClaimBonus The total number of tokens recieved by each user who claims during a particular epoch.
+ * @property {bigint} index The index of a particular epoch.
+ * @property {number} inDays Length of epoch formatted in total number of days
+ */
+
+const startupAssertion = (arg, keyName) =>
+  assert(
+    arg,
+    `Contract has been started without required property: ${keyName}.`,
+  );
```

Comment:
Zoe will check `privateArgs` for you if you declare `privateArgsShape`:

```js
import { M, matches, mustMatch } from '@endo/patterns';
import { AmountShape, PurseShape } from '@agoric/ertp/src/typeGuards.js';

export const privateArgsShape = {
  purse: PurseShape,
  timer: M.remotable('Timer'),
  // etc.
};
```

Reply:
thanks for the input. IOU

- [ ] an updated `prepare.js` file containing `privateArgsShape`.